### PR TITLE
Remove Body mixin; document Request and Response APIs instead

### PR DESCRIPTION
**Pull request by Elchi3**
This PR removes https://developer.mozilla.org/en-US/docs/Web/API/Body and documents the mixin as

- https://developer.mozilla.org/en-US/docs/Web/API/Request and
- https://developer.mozilla.org/en-US/docs/Web/API/Response

BCD: https://github.com/mdn/browser-compat-data/pull/10958

### CODE REVIEW

File: files/en-us/web/api/abortcontroller/abort/index.html [hunk 1]
```diff
@@ -13,7 +13,7 @@
 ---
 <div>{{APIRef("DOM")}}</div>
 
-<p>The <strong><code>abort()</code></strong> method of the {{domxref("AbortController")}} interface aborts a DOM request before it has completed. This is able to abort <a href="/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch">fetch requests</a>, the consumption of any response {{domxref("Body")}}, or streams.</p>
+<p>The <strong><code>abort()</code></strong> method of the {{domxref("AbortController")}} interface aborts a DOM request before it has completed. This is able to abort <a href="/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch">fetch requests</a>, the consumption of any response bodies, or streams.</p>
```
Comment (foolip) [inline files/en-us/web/api/abortcontroller/abort/index.html:16]: Good fix, the previous link to `Body` didn't really make sense IMHO.

File: files/en-us/web/api/fetch_api/basic_concepts/index.html [hunk 2]
```diff
@@ -22,7 +22,7 @@
 
 <h2 id="In_a_nutshell">In a nutshell</h2>
 
-<p>At the heart of Fetch are the Interface abstractions of HTTP {{domxref("Request")}}s, {{domxref("Response")}}s, {{domxref("Headers")}}, and {{domxref("Body")}} payloads, along with a {{domxref("WindowOrWorkerGlobalScope/fetch","global fetch")}} method for initiating asynchronous resource requests. Because the main components of HTTP are abstracted as JavaScript objects, it is easy for other APIs to make use of such functionality.</p>
```
Comment (foolip) [inline files/en-us/web/api/fetch_api/basic_concepts/index.html:25]: Oh my, this was bad. A non-existent interface at the heart of Fetch!

File: files/en-us/web/api/fetch_api/using_fetch/index.html [hunk 3]
```diff
@@ -416,14 +416,14 @@ <h2 id="Body">Body</h2>
  <li>{{domxref("FormData")}}</li>
 </ul>
 
-<p>The {{domxref("Body")}} mixin defines the following methods to extract a body (implemented by both {{domxref("Request")}} and {{domxref("Response")}}). These all return a promise that is eventually resolved with the actual content.</p>
+<p>The {{domxref("Request")}} and {{domxref("Response")}} interfaces share the following methods to extract a body. These all return a promise that is eventually resolved with the actual content.</p>
```
Comment (foolip) [inline files/en-us/web/api/fetch_api/using_fetch/index.html:419]: Nice!

File: files/en-us/web/api/request/arraybuffer/index.html [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  - Fetch
+  - Method
+  - Reference
+  - Request
+browser-compat: api.Request.arrayBuffer
+---
+<div>{{APIRef("Fetch")}}</div>
+
+<p>The <strong><code>arrayBuffer()</code></strong> method of the {{domxref("Request")}} interface
+  takes a {{domxref("Request")}} stream and reads it to completion. It returns a promise
```
Comment (foolip) [inline files/en-us/web/api/request/arraybuffer/index.html:16]: What does it mean to read a `Request` to completion? This description makes sense for a `Response`, but for a `Request` I'm actually not sure what this method is for. I guess it return data already set for the `Request` by some other means?

This comment will apply to all of the methods on `Request`.

More generally, the "takes a" wording here seems pretty unhelpful. While one can do `Request.prototype.arrayBuffer.call(myRequest)`, the `this` is typically implied and I'd expect this to be misread as there being a `Request` argument somewhere, which there isn't.

File: files/en-us/web/api/request/arraybuffer/index.html [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+<p>None.</p>
+
+<h3 id="Return_value">Return value</h3>
+
+<p>A promise that resolves with an {{jsxref("ArrayBuffer")}}.</p>
+
+<h2 id="Examples">Examples</h2>
+
+<pre class="brush: js">
+const myArray = new ArrayBuffer(512);
```
Comment (foolip) [inline files/en-us/web/api/request/arraybuffer/index.html:38]:
```suggestion
const myArray = new Uint8Array(10);
```

The indirection of `longInt8View` doesn't really make this example clearer IMHO. In fact by using a `Uint8Array` it's more clear that one can get an `ArrayBuffer` out even if there wasn't one to begin with.

File: files/en-us/web/api/request/body/index.html [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+<h2 id="Examples">Examples</h2>
+
+<pre class="brush: js">
+const request = new Request('/myEndpoint', {
+  method: 'POST',
+  body: 'Hello world'
+});
+
+request.body; // ReadableStream
```
Comment (foolip) [inline files/en-us/web/api/request/body/index.html:37]: I can confirm it is a `ReadableStream` in Safari, the only browser that supports this currently.

File: files/en-us/web/api/request/clone/index.html [hunk 7]
```diff
@@ -15,7 +15,7 @@
 
 <p>The <strong><code>clone()</code></strong> method of the {{domxref("Request")}} interface creates a copy of the current <code>Request</code> object.</p>
 
-<p><code>clone()</code> throws a {{jsxref("TypeError")}} if the response {{domxref("Body")}} has already been used. In fact, the main reason <code>clone()</code> exists is to allow multiple uses of {{domxref("Body")}} objects (when they are one-use only.)</p>
+<p><code>clone()</code> throws a {{jsxref("TypeError")}} if the response body has already been used. In fact, the main reason <code>clone()</code> exists is to allow multiple uses of body objects (when they are one-use only.)</p>
```
Comment (foolip) [inline files/en-us/web/api/request/clone/index.html:18]:
```suggestion
<p><code>clone()</code> throws a {{jsxref("TypeError")}} if the request body has already been used. In fact, the main reason <code>clone()</code> exists is to allow multiple uses of body objects (when they are one-use only.)</p>
```

File: files/en-us/web/api/response/index.html [hunk 8]
```diff
@@ -45,52 +49,35 @@ <h2 id="Properties">Properties</h2>
...
- <dd>A simple getter exposing a {{domxref("ReadableStream")}} of the body contents.</dd>
- <dt>{{domxref("Body.bodyUsed")}} {{readonlyInline}}</dt>
- <dd>Stores a {{domxref("Boolean")}} that declares whether the body has been used in a response yet.</dd>
-</dl>
-
 <h2 id="Methods">Methods</h2>
 
 <dl>
+ <dt>{{domxref("Response.arrayBuffer()")}}</dt>
+ <dd>Takes a {{domxref("Response")}} stream and reads it to completion. It returns a promise that resolves with an {{domxref("ArrayBuffer")}}.</dd>
```
Comment (foolip) [inline files/en-us/web/api/response/index.html:56]: The "Takes a" wording shows up here too, and is more misleading without the examples that follow on the pages where the wording is used.

File: files/en-us/web/api/request/text/index.html [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+<pre class="brush: js">
+const text = "Hello world";
+
+const request = new Request('/myEndpoint', {
+  method: 'POST',
+  body: text
+});
+
+request.formData().then(function(text) {
```
Comment (foolip) [inline files/en-us/web/api/request/text/index.html:44]:
```suggestion
request.text().then(function(text) {
```

Right?

File: files/en-us/web/api/request/index.html [hunk 10]
```diff
@@ -51,39 +55,25 @@ <h2 id="Properties">Properties</h2>
...
- <dt>{{domxref("Body.formData()")}}</dt>
+ <dt>{{domxref("Request.clone()")}}</dt>
+ <dd>Creates a copy of the current <code>Request</code> object.</dd><dt
+ >{{domxref("Request.formData()")}}</dt>
  <dd>Returns a promise that resolves with a {{domxref("FormData")}} representation of the request body.</dd>
- <dt>{{domxref("Body.json()")}}</dt>
- <dd>Returns a promise that resolves with a {{domxref("JSON")}} representation of the request body.</dd>
- <dt>{{domxref("Body.text()")}}</dt>
+ <dt>{{domxref("Request.json()")}}</dt>
+ <dd>Returns a promise that resolves with a {{jsxref("JSON")}} representation of the request body.</dd>
```
Comment (foolip) [inline files/en-us/web/api/request/index.html:70]: This is not pedantically correct. JSON is bytes or a string, but `request.json()` returns to some value that is the result of calling `JSON.parse(bodyBytes)`. Maybe pedantically correct isn't the best tradeoff for all documentation thought :)

File: files/en-us/web/api/request/index.html [hunk 11]
```diff
@@ -27,6 +27,10 @@ <h2 id="Constructor">Constructor</h2>
 <h2 id="Properties">Properties</h2>
 
 <dl>
+ <dt>{{domxref("Request.body")}} {{readonlyInline}}</dt>
+ <dd>A {{domxref("ReadableStream")}} of the body contents.</dd>
+ <dt>{{domxref("Request.bodyUsed")}} {{readonlyInline}}</dt>
+ <dd>Stores a {{domxref("Boolean")}} that declares whether the body has been used in a response yet.</dd>
```
Comment (foolip) [inline files/en-us/web/api/request/index.html:33]:
```suggestion
 <dd>Stores a {{domxref("Boolean")}} that declares whether the body has been used yet.</dd>
```

I'm actually not sure if this is right, but "response" was wrong.

**Review by foolip — COMMENTED** [review]
Looks great! I have some comments but I won't be the one to approve.

**Review by sideshowbarker — APPROVED** [review]

**Comment by sideshowbarker** [comment]
well dang — pushed the wrong buttons and merged this prematurely

Not sure what’s best to do now, as far as getting the commit suggestions applied.

I guess the options are:

* A. Revert the commit, and raise a new PR, and re-start the review.
* B. Don’t revert the commit, but just a raise a new PR, with the commit suggestions applied, and then I guess with the other review comments re-raised there.

File: files/en-us/web/api/request/arraybuffer/index.html (hunk 4 above)
Comment (Elchi3) [reply files/en-us/web/api/request/arraybuffer/index.html:16]: How about this?

> The <strong><code>arrayBuffer()</code></strong> method of the {{domxref("Request")}} interface reads the request body and returns a promise that resolves with an {{jsxref("ArrayBuffer")}}.

**Comment by Elchi3** [comment]
I opened https://github.com/mdn/content/pull/6468 as the follow-up PR.

**Pull request merged**